src/test/fs_tests.cpp

```diff
@@ -69,6 +70,21 @@ BOOST_AUTO_TEST_CASE(fsbridge_fstream)
         BOOST_CHECK_EQUAL(tmpfile1, fsbridge::AbsPathJoin(tmpfile1, ""));
         BOOST_CHECK_EQUAL(tmpfile1, fsbridge::AbsPathJoin(tmpfile1, {}));
     }
+    {
+        fs::path p1 = GetUniquePath(tmpfolder);
+        fs::path p2 = GetUniquePath(tmpfolder);
+        fs::path p3 = GetUniquePath(tmpfolder);
+
+        // Ensure that the parent path is always the same.
+        BOOST_CHECK_EQUAL(tmpfolder, p1.parent_path());
+        BOOST_CHECK_EQUAL(tmpfolder, p2.parent_path());
+        BOOST_CHECK_EQUAL(tmpfolder, p3.parent_path());
+
+        // Ensure that generated paths are actually different.
+        BOOST_CHECK(p1 != p2);
```

Comment:
I have left this intentionally very simple. One can certainly come up with many different ways to make this more robust but I would like to solicit for feedback for this.